### ISSUE TITLE
Crowd Control Effects [Algalon]

### ISSUE THREAD

**Issue by pascagabriel** [dalaranwow/dalaran-wow#1630]
You are currently unable to control your character for around 0.2-0.3 more seconds after a Crowd Control effect fades from you. This allows for unintended gameplay "tricks" such as Gouge>Sap when combined with the fact that CC effects cause you to lose control of your character prior to actually being affected by them.
 
Also I think it would be appreciated if Spells had an appropriate spell delay implemented. It is impossible to reactively reply to some, such as Shadowstep (Shadowstepping another's Rogue Shadowstep), Vanishing instant spells(can be done if predicted but the margin to do so is extreemely low and Vanish has to be cast prior to the spell being vanished, so that entering stealth and the spell landing occur at the same time), Vanishing missile-type spells as they hit you (has to be done prior to the missile being ~8+ yards away or you will be affected by it), Shadow Word: Deathing instant cast CC unless predicted (with the exception of missile-type instant cast, e.g. Scatter Shot, but then again the same principle as Vanishing them Applies, making reactively SW:D-ing a scatter from less than 10y impossible). 

One more reason spell delay needs to be implemented is to allow two opposing players to cast a spell at the same time. This is impossible at the moment, making the game resemble chess, where each player takes turns to make a move.

Another issue present atm is CC effects causing you to lose control of your character around 0.1-0.2s before you actually get the debuff, which makes faking SW:D as a mage extremely easy, as the priest is forced to use his SW:D around halfway through the Polymorph cast time or he will not have another chance. This intersects with missile-type cc affecting you before it actually lands, such as death coil, scatter shot, Avenger's Shield, Silencing Shot, etc.

**Comment by Lothloryen** [comment]
Spell delay system implemented on Algalon, it will be live with the next restart, in 10-15 minutes.

Performed many tests and I think it works quite well, rogues can sap each other at the same time, 2 mages polimorph each others...etc.

Feedback will be very welcome.

Thank you.

**Comment by pascagabriel** [comment]
I've tested this and it seems to be working correctly. The delay could be around 0.05-0.1s loner but that can be risky as I've seen that while works perfectly for people with under 100 latency, it creates problems for anyone else. In my opinion right now it's fine. Thank you!

**Comment by pascagabriel** [comment]
Also noticed something that's related and has been present for a while now. Sapping someone as they enter combat causes the target to "immune" the sap. No error message, no line in the combat log, global goes off and energy is consumed but the sap does not land.

**Comment by Rushor** [comment]
The spell delay system is not implemented on PTR, only on Live. I cannot change anything here so far
